src/commands/git-metadata/git.ts

```diff
@@ -25,6 +24,25 @@ export const newSimpleGit = async (): Promise<simpleGit.SimpleGit> => {
   return simpleGit.gitP(options)
 }
 
+// Returns a configured SimpleGit or undefined
+// in the case that the user does not have git installed
+export const newSimpleGitOrFail = async (): Promise<simpleGit.SimpleGit> => {
```

Comment:
Could this function be factorized with `newSimpleGit` as it shares most of its implementation?

Reply:
I'm not really sure how to refactor this function because the original logical swallows the exceptions. newSimpleGit() will return a GitClient even if git is not installed, and subsequent calls to the gitclient will throw an error. In upload.ts, this is handled by checking if the return value of getCommitInfo is undefined, and getCommitInfo returns undefined by wrapping everything in a try catch and just returning undefined if an exception is thrown. To me, it makes more sense to fail early. Do you have any recommendations here?

Reply:
After a discussion with @jybp I've refactored the Upload command to use the library.